### ISSUE TITLE
Sort order missing on Downloadable Product Links and Sample Columns

### ISSUE THREAD

**Issue by mahesh-rajawat** [magento/magento2#21278]
### Preconditions (*)

1. Magento 2.2
2. Magento 2.3.0

### Steps to reproduce (*)

1. I need to add a column after Links Title column programatically.
2. I have created a plugin of 
`\Magento\Downloadable\Ui\DataProvider\Product\Form\Modifier\Links::modifyMeta()`
```php
namespace Webkul\Test\Plugin\Magento\Downloadable\Ui\DataProvider\Product\Form\Modifier;
use Magento\Catalog\Model\Locator\LocatorInterface;
use Magento\Downloadable\Ui\DataProvider\Product\Form\Modifier\Composite;
use Magento\Framework\Stdlib\ArrayManager;
use Magento\Ui\Component\Container;
use Magento\Ui\Component\Form;
/**
 * Links class
 */
class Links 
{
    /**
     *
     * @param \Webkul\Test\Helper\Data $helper
     */
    public function __construct(
        LocatorInterface $locator,
        ArrayManager $arrayManager,
        \Webkul\Test\Helper\Data $helper
    ) {
        $this->helper = $helper;
        $this->locator = $locator;
        $this->arrayManager = $arrayManager;
    }
    /**
     * @see \Magento\Downloadable\Ui\DataProvider\Product\Form\Modifier\Links::modifyMeta()
     */
    public function afterModifyMeta (
        \Magento\Downloadable\Ui\DataProvider\Product\Form\Modifier\Links $subject,
        $result
    ) {
            $recordPath = $this->arrayManager->findPath('record', $result, null, 'children');
            $result = $this->arrayManager->merge(
                $recordPath.'/children',
                $result,
                array_merge(
                    [
                        'resolution' => $this->getResolutionColumn()
                    ]
                )
                
            );
        return $result;
    }
    /**
     * @return array
     */
    protected function getResolutionColumn()
    {
        $titleContainer['arguments']['data']['config'] = [
            'componentType' => Container::NAME,
            'formElement' => Container::NAME,
            'component' => 'Magento_Ui/js/form/components/group',
            'label' => __('Resolution'),
            'showLabel' => false,
            'dataScope' => '',
            'sortOrder' => 15,
        ];
        $titleField['arguments']['data']['config'] = [
            'formElement' => Form\Element\Input::NAME,
            'componentType' => Form\Field::NAME,
            'dataType' => Form\Element\DataType\Text::NAME,
            'dataScope' => 'resolution',
            'validation' => [
                'required-entry' => true,
            ],
        ];
        return $this->arrayManager->set('children/resolution', $titleContainer, $titleField);
    }
}
```
3.  After applying this code with **sortOrder 15** on this custom column the links table columns misplaced 

### Expected result (*)

1.  This custom column must be insert after Title Column
2.

### Actual result (*)

1. 
![download](https://user-images.githubusercontent.com/32479280/52877335-afb51a80-317f-11e9-83fa-fa43069f878b.png)

2.

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@maheshWebkul721 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @shikhamis11. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by mahesh-rajawat** [comment]
@magento-engcom-team issue can be reproduced.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98301`, `MAGETWO-98302` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for your report. 
The issue has been fixed in magento/magento2#21279 by @maheshWebkul721 in 2.3-develop branch
Related commit(s):
 - [c9a9d6406aaa5a4d68e2ed35abe5f847c394285c](https://github.com/magento/magento2/commit/c9a9d6406aaa5a4d68e2ed35abe5f847c394285c)
 - [92d2a3c1e5a2f71c67b1335b7d909136e106b554](https://github.com/magento/magento2/commit/92d2a3c1e5a2f71c67b1335b7d909136e106b554)
 - [03f0c3aa6c5e9bc2207c197b54e89ae727e15342](https://github.com/magento/magento2/commit/03f0c3aa6c5e9bc2207c197b54e89ae727e15342)
 - [81a4de89eee31ebf68cdabf3937437470f6661a0](https://github.com/magento/magento2/commit/81a4de89eee31ebf68cdabf3937437470f6661a0)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @maheshWebkul721. Thank you for your report. 
The issue has been fixed in magento/magento2#21662 by @mage2pratik in 2.2-develop branch
Related commit(s):
 - [f80d38218d0d42a297fe6f6727f5d63191093fad](https://github.com/magento/magento2/commit/f80d38218d0d42a297fe6f6727f5d63191093fad)
 - [b7ad0754b9f78510ec1f47ddf866d70ed05f91fc](https://github.com/magento/magento2/commit/b7ad0754b9f78510ec1f47ddf866d70ed05f91fc)
 - [a9016e9a58187fa8bb81cfecd6e0bc5bfdd6ce90](https://github.com/magento/magento2/commit/a9016e9a58187fa8bb81cfecd6e0bc5bfdd6ce90)

The fix will be available with the upcoming 2.2.9 release.
